zrml/prediction-markets/src/lib.rs

```diff
@@ -1934,20 +1937,30 @@ mod pallet {
                 let market = T::MarketCommons::market(market_id)?;
                 mutation(market_id, market)?;
             }
+
             MarketIdsPerCloseBlock::<T>::remove(&now);
 
+            // If we are at genesis the timestamp is 0. No market can exist, we skip the evaluation.
+            // Without this check, new chains starting from genesis will hang up, since the loop
+            // below will run over an interval of 0 to the current time frame.
             let current_time_frame = Self::calculate_time_frame_of_moment(T::MarketCommons::now());
+            if current_time_frame == 0u64 {
+                return Ok(());
+            }
```

Comment:
The API of `pallet-timestamp` states (https://github.com/paritytech/substrate/blob/aacc3e99c0750c3e6c851a8270dc2c4eebca3290/frame/timestamp/src/lib.rs#L297-L299):

> On genesis the time returned is not valid. 

So strictly speaking, this check relies on undefined behavior. Unless there's an argument against this, I think you should just check if the current block number is zero instead.

Reply:
That's a great suggestion!